llama_stack/distribution/server/auth_providers.py

```diff
@@ -339,6 +339,134 @@ async def close(self):
             self._client = None
 
 
+class OpenShiftOAuthAuthProviderConfig(BaseModel):
```

Comment:
The 'OAuth' in the name here seems misleading to me as the actual implementation does not rely on any oauth related specification as far as I can see. OpenshiftAuthProviderConfig would be clearer.

Would the use case be met by using the standard kubernetes SelfSubjecReview[1]? Can openshift not issue jwt tokens for users?

[1] https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/self-subject-review-v1/